src/EditorFeatures/Core/Implementation/IntelliSense/ModelComputation.cs

```diff
@@ -143,22 +143,23 @@ public virtual void Stop()
             // then we don't need to notify as a later task will do so.
             _notifyControllerTask = Task.Factory.ContinueWhenAll(
                 new[] { _notifyControllerTask, nextTask },
-                tasks =>
+                async tasks =>
+                {
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_stopCancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.